sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ExternalSource.scala

```diff
@@ -50,29 +79,19 @@ object ExternalSource extends Logging {
   /** If the file format is supported as an external source */
   def isSupportedFormat(format: FileFormat): Boolean = {
     if (hasSparkAvroJar) {
-      format match {
-        case _: AvroFileFormat => true
-        case _ => false
-      }
+      ShimLoader.avroProvider.isSupportedFormat(format)
```

Comment:
Add a lazy val to `object ExternalSource`

```scala
lazy val avroProvider = ShimLoader.newAvroProvider()
```

and here and elsewhere simply call methods on the lazy val